lib/Wikia/src/Persistence/User/Preferences/PreferencePersistence.php

```diff
@@ -228,9 +232,39 @@ private function findWikisWithLocalPreference( ReverseLookupApi $api, $preferenc
 		return $wikiList;
 	}
 
-	private function findUsersWithGlobalPreference( ReverseLookupApi $api, $preferenceName, $value = null ) {
-		$userList = $api->findUsersWithGlobalPreference( $preferenceName, $value );
-		return $userList;
+	/**
+	 * Calls user-preference /reverse-lookup/global/{preferenceName}/users endpoint to get an array of string userIds
+	 * of users with $preferenceName set to $value (or set at all if $value is null)
+	 * Handles paginating API calls for $numUsersToRetrieve > 10000
+	 * @param ReverseLookupApi $api
+	 * @param string $preferenceName
+	 * @param string|null $value
+	 * @param int|null $numUsersToRetrieve
+	 * @param int|null $userIdContinue
+	 * @return array|string[] - userId
+	 * @throws ApiException
+	 */
+	private function findUsersWithGlobalPreference(ReverseLookupApi $api, $preferenceName, $value = null, $numUsersToRetrieve = 1000, $userIdContinue = null ) {
+		if ( $numUsersToRetrieve <= PreferencePersistence::REVERSE_LOOKUP_GLOBAL_USERS_MAX_LIMIT ) {
+			return $api->findUsersWithGlobalPreference( $preferenceName, $value, $numUsersToRetrieve, $userIdContinue );
+		}
+		else {
+			// pagination for API calls
+			$userList = [];
+			$userIdContinue = intval($userIdContinue);
+			while ( count($userList) < $numUsersToRetrieve ){
```

Comment:
If the numUsersToRetrieve is larger than the number of users in the database this loop will never finish

Reply:
I'll remove pagination altogether and just put a hard limit on $numUsersToRetrieve, another PR :)